### ISSUE TITLE
[PATCH] [TEST] problem with arg handling in soap service

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7223]
|<img alt="waterbug's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| waterbug reported|
|-|-|
|Trac ID|trac#488|
|Type|defect|
|Created|2004-02-05 01:14:12Z|

Attachments:

* [soapbug.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ee1/ee16ee0fa7f21a4101793582e1cf4b24fef9f948/7ceacbdca97aa00468c5d8f0899810c4f163044c.txt) (4596 bytes) - added by waterbug on 2004-02-05 01:14:12Z - 
* [soap.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ee1/ee16ee0fa7f21a4101793582e1cf4b24fef9f948/6116b3c11e206bc26c7f012f5c52774ec26e0bdf.patch) (1457 bytes) - added by waterbug on 2004-02-05 10:05:04Z - 
* [test_soap.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ee1/ee16ee0fa7f21a4101793582e1cf4b24fef9f948/8be6ab43bfe4e2918a1bb8cc80ea439bdde4deaa.patch) (910 bytes) - added by waterbug on 2004-02-05 10:21:16Z - 
* [soap.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ee1/ee16ee0fa7f21a4101793582e1cf4b24fef9f948/1c83d569c518693bd90c8b543507ad6137f49de5.patch) (606 bytes) - added by radix on 2004-02-20 16:17:07Z - 
* [test_soap.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ee1/ee16ee0fa7f21a4101793582e1cf4b24fef9f948/1d2b8a0d5e8e25ea30de09b45946fc496f429868.patch) (910 bytes) - added by radix on 2004-02-20 16:17:31Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__488 488
type__defect defect
reporter__waterbug waterbug
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1075943652000000 1075943652000000
changetime__1471142772094123 1471142772094123
version__ 
owner__ 
cc__radix cc__itamarst cc__waterbug
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="waterbug's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|waterbug commented|
|-|-|

```
#!html
<pre>
Here's a patch for t.w.soap.py that fixes the problem.
(Very odd -- this never would have worked the way it
was ... must have got changed by mistake or something.)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="waterbug's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|waterbug commented|
|-|-|

```
#!html
<pre>
And here's a patch for the unit test,
t.test.test_soap.py (which had the
same error).
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Hmm... I just tried running doc/examples/soap.py and did what it said in the
docstring, and it's working fine.

>>> import SOAPpy
>>> p = SOAPpy.SOAPProxy('http://localhost:8080/')
>>> p.add(a=1)
1
>>> p.add(a=1, b=3)
4
>>> p.echo([1,2])
[1, 2]

Dunno what the problem could be.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
Just noticed steve sent me a message but not the bug. his patch was incorrect,
I've attached the real one (that he sent me).
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
and the test patch
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
The problem seemed to be with changing SOAPpy interfaces from 0.10 to 0.11. I
checked in a fix which hopefully will work with both. Could you please test to
see if it's working?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="waterbug's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|waterbug commented|
|-|-|

```
#!html
<pre>
The example (doc/examples/soap.py) works, but teh test
(twisted/test/test_soap.py) fails:

[waterbug@bigboote Twisted]$ trial twisted/test/test_soap.py 
S
/home/waterbug/Twisted/twisted/test/test_soap.py:75:
exceptions.DeprecationWarning: IPv4Address.__getitem__ is deprecated.  Use
attributes instead.
F
===============================================================================
SKIPPED: testErrors (twisted.test.test_soap.SOAPTestCase)
-------------------------------------------------------------------------------
Not yet implemented
===============================================================================
FAILURE: testResults (twisted.test.test_soap.SOAPTestCase)
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/waterbug/Twisted/twisted/trial/unittest.py", line 252, in _runPhase
    stage(*args, **kwargs)
  File "/home/waterbug/Twisted/twisted/trial/unittest.py", line 274, in _main
    self.runner(self.method)
  File "/home/waterbug/Twisted/twisted/trial/runner.py", line 104, in runTest
    method()
  File "/home/waterbug/Twisted/twisted/test/test_soap.py", line 100, in testResults
    {"a": "c"})
  File "/home/waterbug/Twisted/twisted/trial/unittest.py", line 127, in
failUnlessEqual
    raise FailTest, (msg or '%r != %r' % (first, second))
FailTest: &lt;bound method structType._asdict of &lt;SOAPpy.Types.structType Result at
1084167980>: {'a': 'c'}> != {'a': 'c'}
-------------------------------------------------------------------------------
Ran 2 tests in 0.111s

FAILED (failures=1, skips=1)
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set status to `closed`|
|-|-|

It seems it has been solved a long time ago in [14782].

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_11"></a>|GitHub commented|
|-|-|

In changeset 2472c6fc99386aacb6eb430aab52a38f67c69b61
```
#!CommitTicketReference repository="" revision="2472c6fc99386aacb6eb430aab52a38f67c69b61"
Merge pull request #488 from twisted/8747-rodrigc-zippath-py3

Author: hawkowl, rodrigc
Reviewer: markrwilliams
Fixes: #8747

Fix test_zippath on Windows Python 3
```
